### PR TITLE
fix: always spread `arguments` in `super` call with implicit args

### LINKED ISSUE #310: Wrong super keyword compilation.
decaffeinate version 2.17.1

```coffeescript
class C extends B
  constructor: (x) ->
    super

  fn: (x) ->
    super
```
```javascript
class C extends B {
  constructor(x) {
    super();
  }

  fn(x) {
    return super.fn();
  }
}
```
ES6 code should call super with `arguments` in this examples.

### PR DESCRIPTION
**Pull request by eventualbuddha**
Fixes #310

### CODE REVIEW

File: src/stages/normalize/patchers/FunctionApplicationPatcher.js [hunk 1]
```diff
@@ -23,6 +23,12 @@ export default class FunctionApplicationPatcher extends NodePatcher {
       return;
     }
 
+    if (this.isImplicitSuper()) {
+      this.log(this.fn);
```
Comment (alangpierce) [inline src/stages/normalize/patchers/FunctionApplicationPatcher.js]: Maybe you left this in by mistake?

**Comment by alangpierce** [comment]
LGTM. Too bad the AST format isn't more direct here.

File: src/stages/normalize/patchers/FunctionApplicationPatcher.js (hunk 1 above)
Comment (eventualbuddha) [inline src/stages/normalize/patchers/FunctionApplicationPatcher.js]: Yep, thanks!

**Comment by eventualbuddha** [comment]
> Too bad the AST format isn't more direct here.

Well, we do control that 😉

**Pull request merged**